### PR TITLE
Update 2.6 base centos and update libbpf-devel

### PR DESCRIPTION
**Pull request by yxun**

### CODE REVIEW

File: docker/maistra-builder_2.6.Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # hadolint ignore=DL3006
-FROM registry.access.redhat.com/ubi9/ubi:9.3-1476
+FROM quay.io/centos/centos:stream9
```
[the remark this answers is not in the document]
Comment (yxun) [reply docker/maistra-builder_2.6.Dockerfile:16]: Yes, like line 61 

>  libxcrypt-compat-0:4.4.18-3.el9

File: docker/maistra-builder_2.6.Dockerfile [hunk 2]
```diff
@@ -55,23 +56,20 @@ RUN dnf -y upgrade --refresh && dnf -y install --setopt=install_weak_deps=False
...
 # Install libbpf-devel from centos stream
-# Do not run "dnf -y upgrade --refresh" again. Avoid conflicts between ubi and centos dnf upgrade.
+# Do not run "dnf -y upgrade --refresh" again. Avoid conflicts between dnf upgrades.
 RUN set -eux; \
     dnf -y install --setopt=install_weak_deps=False \
-    "https://mirror.stream.centos.org/9-stream/BaseOS/$(uname -m)/os/Packages/libzstd-1.5.0-2.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/AppStream/$(uname -m)/os/Packages/libzstd-devel-1.5.0-2.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/AppStream/$(uname -m)/os/Packages/elfutils-libelf-devel-0.189-3.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/CRB/$(uname -m)/os/Packages/libbpf-devel-1.2.0-1.el9.$(uname -m).rpm"
+    "https://mirror.stream.centos.org/9-stream/CRB/$(uname -m)/os/Packages/libbpf-devel-1.1.0-2.el9.$(uname -m).rpm"
```
[the remark this answers is not in the document]
Comment (yxun) [reply docker/maistra-builder_2.6.Dockerfile:72]: You are right. I will install it with --enablerepo=crb

File: docker/maistra-builder_2.6.Dockerfile [hunk 3]
```diff
@@ -55,23 +56,15 @@ RUN dnf -y upgrade --refresh && dnf -y install --setopt=install_weak_deps=False
...
-# Install libbpf-devel from centos stream
-# Do not run "dnf -y upgrade --refresh" again. Avoid conflicts between ubi and centos dnf upgrade.
-RUN set -eux; \
-    dnf -y install --setopt=install_weak_deps=False \
-    "https://mirror.stream.centos.org/9-stream/BaseOS/$(uname -m)/os/Packages/libzstd-1.5.0-2.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/AppStream/$(uname -m)/os/Packages/libzstd-devel-1.5.0-2.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/AppStream/$(uname -m)/os/Packages/elfutils-libelf-devel-0.189-3.el9.$(uname -m).rpm" \
-    "https://mirror.stream.centos.org/9-stream/CRB/$(uname -m)/os/Packages/libbpf-devel-1.2.0-1.el9.$(uname -m).rpm"
+    sudo autoconf automake cmake unzip wget xz procps dnf-plugins-core \
+    libbpf-devel-2:1.1.0-2.el9
```
Comment (jwendell) [inline docker/maistra-builder_2.6.Dockerfile:67]: Why are you specifying a version here? Why not just `libbpf-devel`?

Comment (jwendell) [reply docker/maistra-builder_2.6.Dockerfile]: We should not rely on fedora's packages to build or test our software. We should use the builder container for that, and we should make sure the builder container has all the necessary packages with the correct versions to build&test our software.

**Review by jwendell — APPROVED** [review]

**Pull request merged**